builtin/common/misc_helpers.lua

```diff
@@ -238,6 +238,52 @@ function math.sign(x, tolerance)
 end
 
 --------------------------------------------------------------------------------
+-- Video enums and packing function
+
+local blend_ebf = { -- E_BLEND_FACTOR in irrlicht
+	zero                  = 0, -- src & dest (0, 0, 0, 0)
+	one                   = 1, -- src & dest (1, 1, 1, 1)
+	dst_color             = 2, -- src (destR, destG, destB, destA)
+	dst_color_inv         = 3, -- src (1-destR, 1-destG, 1-destB, 1-destA)
+	src_color             = 4, -- dest (srcR, srcG, srcB, srcA)
+	src_color_inv         = 5, -- dest (1-srcR, 1-srcG, 1-srcB, 1-srcA)
+	src_alpha             = 6, -- src & dest (srcA, srcA, srcA, srcA)
+	src_alpha_inv         = 7, -- src & dest (1-srcA, 1-srcA, 1-srcA, 1-srcA)
+	dst_alpha             = 8, -- src & dest (destA, destA, destA, destA)
+	dst_alpha_inv         = 9, -- src & dest (1-destA, 1-destA, 1-destA, 1-destA)
+	src_alpha_saturate    = 10,-- src (min(srcA, 1-destA), ...)
+}
+
+local blend_emfn = { -- E_MODULATE_FUNC in irrlicht
+	["1x"]    = 1,
+	["2x"]    = 2,
+	["4x"]    = 4,
+}
+
+local blend_eas = { -- E_ALPHA_SOURCE in irrlicht
+	none         = 0,
+	vertex       = 1,
+	texture      = 2,
+	both         = 3,
+}
+
+function core.pack_texture_blend_func(srcFact, dstFact, modulate, alphaSource)
+	local srcFact     = assert(blend_ebf[srcFact], "invalid srcFact")
```

Comment:
@sfan5 Do you think it's a good idea to force people to use the name instead of the value directly?

how about:
`assert(blend_ebf[srcFact] or type(srcFact) == "number" and srcFact, "invalid srcFact")`